pinot-core/src/main/java/org/apache/pinot/core/operator/filter/BitmapBasedFilterOperator.java

```diff
@@ -66,8 +67,9 @@ protected FilterBlock getNextBlock() {
 
     int[] dictIds = _exclusive ? _predicateEvaluator.getNonMatchingDictIds() : _predicateEvaluator.getMatchingDictIds();
     int numDictIds = dictIds.length;
-    // NOTE: PredicateEvaluator without matching/non-matching dictionary ids should not reach here.
-    Preconditions.checkState(numDictIds > 0);
+    if (numDictIds == 0) {
```

Comment:
Why this change?

Reply:
Without this precondition fails, I checked with Jackie who added this condition he mentioned it should be fine to remove this. Reason could be because I enabled regexp operator to use bitmap based filter when FST index is available.